src/odb/src/3dblox/dbvWriter.cpp

```diff
@@ -0,0 +1,251 @@
+      region_node["side"] = "internal_ext";
+      break;
+  }
+  if (auto layer = region->getLayer(); layer != nullptr) {
+    region_node["layer"] = layer->getName();
+  }
+  YAML::Node coords_node = region_node["coords"];
+  writeCoordinates(coords_node, region->getBox(), db);
+}
+
+void DbvWriter::writeExternal(YAML::Node& external_node,
+                              odb::dbChip* chiplet,
+                              odb::dbDatabase* db)
+{
+  BaseWriter::writeLef(external_node, db, chiplet);
+  if (db->getChip()->getBlock() != nullptr) {
+    BaseWriter::writeDef(external_node, db, chiplet);
+  }
+}
+
+void DbvWriter::writeCoordinates(YAML::Node& coords_node,
+                                 const odb::Rect& rect,
+                                 odb::dbDatabase* db)
+{
+  const double u = db->getDbuPerMicron();
+  YAML::Node c0, c1, c2, c3;
+  c0.SetStyle(YAML::EmitterStyle::Flow);
+  c1.SetStyle(YAML::EmitterStyle::Flow);
+  c2.SetStyle(YAML::EmitterStyle::Flow);
+  c3.SetStyle(YAML::EmitterStyle::Flow);
+  c0.push_back(rect.xMin() / u);
+  c0.push_back(rect.yMin() / u);
+  c1.push_back(rect.xMax() / u);
+  c1.push_back(rect.yMin() / u);
+  c2.push_back(rect.xMax() / u);
+  c2.push_back(rect.yMax() / u);
+  c3.push_back(rect.xMin() / u);
+  c3.push_back(rect.yMax() / u);
+  coords_node.push_back(c0);
+  coords_node.push_back(c1);
+  coords_node.push_back(c2);
+  coords_node.push_back(c3);
+}
+
+void DbvWriter::writeChipDependencies(YAML::Node& header_node,
+                                      const ChipletNode* node)
+{
+  std::unordered_set<std::string> included;
+
+  for (auto child : node->children) {
+    auto* child_chip = child->chip;
+    if (child_chip->getChipType() == odb::dbChip::ChipType::HIER) {
+      // Add the child's .3dbx file include
+      std::string child_3dbx = std::string(child_chip->getName()) + ".3dbx";
+      included.insert(child_3dbx);
+      // TODO: Call dbxWriter::writeChiplet(child_chip)
+    }
+  }
+  if (!included.empty()) {
+    YAML::Node includes_node = header_node["include"];
+    for (const auto& include : included) {
+      includes_node.push_back(include);
+    }
+  }
+}
+
+void DbvWriter::writeChipletToFile(const std::string& filename,
+                                   odb::dbChip* chiplet,
+                                   odb::dbDatabase* db,
+                                   ChipletNode* node)
+{
+  YAML::Node root;
+  YAML::Node header_node = root["Header"];
+  writeHeader(header_node, db);
+
+  writeChipDependencies(header_node, node);
+
+  YAML::Node chiplets_node = root["ChipletDef"];
+  for (auto dependecy : node->children) {
+    YAML::Node chiplet_node = chiplets_node[dependecy->chip->getName()];
+    writeChipletInternal(chiplet_node, dependecy->chip, db);
+  }
+
+  writeYamlToFile(filename, root);
+}
+
+void DbvWriter::writeChiplet(const std::string& base_filename,
+                             odb::dbDatabase* db,
+                             odb::dbChip* top_chip)
```

Comment:
You could get one from the other. Only one is sufficient, either db or top_chip.